pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -502,7 +537,7 @@ def default_model_config(self) -> dict:
                 "m": 200,
```

Comment:
Wrap these existing ones in the HSGPKwargs then

Reply:
Seeing that some of the serialization methods work for the `Fourier` classes

For instance, 
```python
from pymc_marketing.mmm import YearlyFourier

fourier = YearlyFourier(n_order=3)
fourier.model_dump()
```

However, there is an issue with 
```python
fourier.json()
```

Reply:
From what I have read online, it seems `fourier.model_dump()` is the way to go. The issue comes because there is no canonical way to convert `Prior` to JSON. Let me keep investigating.

Reply:
Solved in https://github.com/pymc-labs/pymc-marketing/pull/809/commits/1bd9627a1578a1e6d1caddce5321dffa8f73077b :)